apis/apps/v1/cluster_types.go

```diff
@@ -0,0 +1,866 @@
+
+	// If ServiceType is LoadBalancer, cloud provider related parameters can be put here.
+	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer.
+	//
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Indicates whether to generate individual Services for each Pod.
+	// If set to true, a separate Service will be created for each Pod in the Cluster.
+	//
+	// +optional
+	PodService *bool `json:"podService,omitempty"`
+}
+
+type ComponentSystemAccount struct {
+	// The name of the system account.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Specifies the password for the account.
+	// The password can be generated based on certain rules or retrieved from a Secret.
+	//
+	// This field is immutable once set.
+	//
+	// +kubebuilder:validation:Required
+	Password SystemAccountPassword `json:"password,omitempty"`
+}
+
+// ClusterComponentConfig represents a config with its source bound.
+type ClusterComponentConfig struct {
+	// The name of the config.
+	//
+	// +optional
+	Name *string `json:"name,omitempty"`
+
+	// The source of the config.
+	ClusterComponentConfigSource `json:",inline"`
+}
+
+// ClusterComponentConfigSource represents the source of a config.
+type ClusterComponentConfigSource struct {
+	// ConfigMap source for the config.
+	//
+	// +optional
+	ConfigMap *corev1.ConfigMapVolumeSource `json:"configMap,omitempty"`
+
+	// TODO: support more diverse sources:
+	// - Config template of other components within the same cluster
+	// - Config template of components from other clusters
+	// - Secret
+	// - Local file
+}
+
+// ClusterService defines a service that is exposed externally, allowing entities outside the cluster to access it.
+// For example, external applications, or other Clusters.
+// And another Cluster managed by the same KubeBlocks operator can resolve the address exposed by a ClusterService
+// using the `serviceRef` field.
+//
+// When a Component needs to access another Cluster's ClusterService using the `serviceRef` field,
+// it must also define the service type and version information in the `componentDefinition.spec.serviceRefDeclarations`
+// section.
+type ClusterService struct {
+	Service `json:",inline"`
+
+	// Extends the ServiceSpec.Selector by allowing the specification of a sharding name, which is defined in
+	// `cluster.spec.shardingSpecs[*].name`, to be used as a selector for the service.
+	// Note that this and the `componentSelector` are mutually exclusive and cannot be set simultaneously.
+	//
+	// +optional
+	ShardingSelector string `json:"shardingSelector,omitempty"`
+
+	// Extends the ServiceSpec.Selector by allowing the specification of a component, to be used as a selector for the service.
+	// Note that this and the `shardingSelector` are mutually exclusive and cannot be set simultaneously.
+	//
+	// +optional
+	ComponentSelector string `json:"componentSelector,omitempty"`
```

Comment:
can these two fields get merged?

for example, there is only a single field `componentSelector`.  if it equals to a name specified in `cluster.spec.shardingSpecs[*].name`, then it's a sharding selector.

Reply:
This way, if I don't use the sharding feature, I won't need to understand the `shardingSelector` field.

Reply:
Done